frontend/src/app/visualization/barchart/barchart.directive.ts

```diff
@@ -372,19 +374,27 @@ export abstract class BarchartDirective
         return rawData;
     }
 
-    getTermFrequencies(series: BarchartSeries<DataPoint>, queryModel: QueryModel): Promise<BarchartSeries<DataPoint>> {
+    getTermFrequencies(series: BarchartSeries<DataPoint>, queryModel: QueryModel) {
         const queryModelCopy = this.queryModelForSeries(series, queryModel);
-        return this.requestSeriesTermFrequency(series, queryModelCopy).then(result =>
-            this.apiService.pollTasks<TermFrequencyResult>(result.task_ids)
-        ).then(res =>
-            this.processSeriesTermFrequency(res, series)
-        ).catch(error => {
-            console.error(error);
-            this.error.emit(`could not load results: ${error.message}`);
-            return series;
+        this.requestSeriesTermFrequency(series, queryModelCopy).then(response => {
```

Comment:
This was already the case, but the logic in this function shouldn't really be handled in the component.

Ideally, the component should just use `this.visualisationService.getTheData(this.destroy$).then(...)` and not care whether the service is fetching the data through polling or in a single request.

But like I said, that was already an issue, so feel free to ignore.

Reply:
I'm not sure I understand. The current setup (once we get to this line) is that we first initiate the tasks, and then, with the task ids, start polling. If we are in this function (`getTermFrequencies`), we are always polling.